### PR TITLE
[WIP] New Qiskit device prototype

### PR DESCRIPTION
**Pull request by austingmhuang**
This is a work in progress. This pull request aims to create a new Qiskit device that uses the new device API in Pennylane. The goal is to be compatible with Qiskit 1.0 and backwards compatible with Qiskit 0.45

### CODE REVIEW

**Comment by austingmhuang** [comment]
Need to update the Plugin Test Matrix before merge.

**Comment by austingmhuang** [comment]
Need to update demos for when upgraded to Qiskit >1.0

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,19 @@
...
+  [(#536)](https://github.com/PennyLaneAI/pennylane-qiskit/pull/536)
+
+* The test suite no longer runs for Qiskit versions below 0.46.
+  [(#536)](https://github.com/PennyLaneAI/pennylane-qiskit/pull/536)
+
+* The ``qiskit.basicaer`` device has been removed because it is not supported for versions of Qiskit above 0.46.
+  [(#546)](https://github.com/PennyLaneAI/pennylane-qiskit/pull/546)
+
+* The IBM quantum devices, ``qiskit.ibmq``, ``qiskit.ibmq.circuit_runner`` and ``qiskit.ibmq.sampler``, have been removed due to deprecations of the IBMProvider and the cloud simulator "ibmq_qasm_simulator".
+  [(#550)](https://github.com/PennyLaneAI/pennylane-qiskit/pull/550)
```
Comment (lillian542) [inline CHANGELOG.md:26]: We should double-check that all of this moves to the 0.38 change-log once the changes from the release are merged in from master

**Review by soranjh — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -30,6 +48,7 @@ This release contains contributions from (in alphabetical order):
 ### Contributors ✍️
 
 This release contains contributions from (in alphabetical order):
+Austin Huang
 
 Utkarsh Azad
 Mashhood Khan
```
Comment (lillian542) [inline CHANGELOG.md:54]:
```suggestion
Utkarsh Azad
Lillian M. A. Frederiksen
Austin Huang
Mashhood Khan
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -30,6 +48,7 @@ This release contains contributions from (in alphabetical order):
 ### Contributors ✍️
 
 This release contains contributions from (in alphabetical order):
+Austin Huang
```
Comment (obliviateandsurrender) [inline CHANGELOG.md:52]:
```suggestion
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,10 +6,28 @@
 
 ### Breaking changes 💔
```
Comment (lillian542) [inline CHANGELOG.md:7]: Should we add a changelog entry for this PR under `Improvements`? Something about how the remote device now uses the Qiskit `Primitive` and `Session` functionality?

**Review by obliviateandsurrender — APPROVED** [review]
:shipit:

**Review by lillian542 — APPROVED** [review]
Wondering if the change-log needs an update before merging, but otherwise looks ready to go!

File: CHANGELOG.md (hunk 4 above)
Comment (austingmhuang) [reply CHANGELOG.md:7]: Hmm... is that an improvement or a feature? I'm surprised `qiskit_session` isn't listed as a feature actually, must have missed adding it in the actual PR. I think I'll add `qiskit_session` as a feature and Primitives as an improvement to the changelog.

File: CHANGELOG.md (hunk 2 above)
Comment (austingmhuang) [reply CHANGELOG.md]: Probably messed this up when resolving a merge conflict 😅

**Pull request merged**